apps/server/src/modules/oauth/controller/api-test/oauth-sso.api.spec.ts

```diff
@@ -83,42 +85,87 @@ describe('OAuth SSO Controller (API)', () => {
 		app = moduleRef.createNestApplication();
 		await app.init();
 		em = app.get(EntityManager);
-
-		const wellKnown = app.get(KeycloakAdministrationService).getWellKnownUrl();
-		axiosMock.onGet(wellKnown).reply(200, {
-			issuer: 'issuer',
-			token_endpoint: 'tokenEndpoint',
-			authorization_endpoint: 'authEndpoint',
-			end_session_endpoint: 'logoutEndpoint',
-			jwks_uri: 'jwksEndpoint',
-		});
 	});
 
 	afterAll(async () => {
-		await cleanupCollections(em);
 		await app.close();
 	});
 
+	afterEach(async () => {
+		await cleanupCollections(em);
+	});
+
 	const setup = async () => {
 		const externalUserId = 'externalUserId';
 		const system: System = systemFactory.withOauthConfig().buildWithId();
 		const school: School = schoolFactory.buildWithId({ systems: [system] });
 		const user: User = userFactory.buildWithId({ externalId: externalUserId, school });
 		const account: Account = accountFactory.buildWithId({ systemId: system.id, userId: user.id });
 
-		await em.persistAndFlush([system, user, school, account]);
+		const targetSystem: System = systemFactory
+			.withOauthConfig()
+			.buildWithId({ provisioningStrategy: SystemProvisioningStrategy.SANIS });
+		const sourceSystem: System = systemFactory
+			.withOauthConfig()
+			.buildWithId({ provisioningStrategy: SystemProvisioningStrategy.ISERV });
+
+		const sourceSchool: School = schoolFactory.buildWithId(
+			{
+				systems: [sourceSystem],
+				officialSchoolNumber: '11111',
+				externalId: 'aef1f4fd-c323-466e-962b-a84354c0e713',
```

Comment:
When you outsource the axios mock, that it should be also expose this id to use it. This hold the id in sync.